packages/elements-material-ui/src/Table/Table.tsx

```diff
@@ -0,0 +1,330 @@
+              checked={row.isSelected}
+              onChange={(e) => onRowSelected(row.original, e.target.checked)}
+            />
+          );
+        },
+      });
+    }
+    return columns as Column<T>[];
+  }, [props.columns, props.expandable]);
+
+  const tableHooks: PluginHook<T>[] = [useFilters, useSortBy];
+  if (props.expandable) {
+    tableHooks.push(useExpanded);
+  }
+  if (props.pagination) {
+    tableHooks.push(usePagination);
+  }
+  if (props.selection) {
+    tableHooks.push(useRowSelect);
+  }
+  tableHooks.push(useFlexLayout);
+
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    rows,
+    prepareRow,
+    state,
+
+    // The page controls ;)
+    page,
+    // canPreviousPage,
+    // canNextPage,
+    pageOptions,
+    pageCount,
+    gotoPage,
+    nextPage,
+    previousPage,
+    // setPageSize,
+
+    // select props
+    toggleAllRowsSelected,
+    isAllRowsSelected,
+    selectedFlatRows,
+    toggleRowSelected,
+  } = useTable(
+    {
+      columns,
+      data: props.data,
+      getRowId: (row: any) => row[props.rowKey],
+      manualSortBy: !!props.onSortChange,
+      manualFilters: !!props.onFilterChange,
+      manualPagination: !!props.onPageChange,
+      manualRowSelectedKey: props.rowKey,
+      pageCount: props.pageCount ?? 0,
+      useControlledState: (state1: any) =>
+        ({
+          ...state1,
+          sortBy: props.sortBy ?? state1.sortBy,
+          filters: props.filters ?? state1.filters,
+          selectedRowIds: props.selectedRowIds ?? state1.selectedRowIds,
+        } as TableState<T> & UseFiltersState<T> & UseSortByState<T> & UseRowSelectState<T>),
+      expandSubRows: false,
+      initialState: {
+        pageIndex: 0,
+        pageSize: props.pageSize ?? 0,
+        selectedRowIds: props.selectedRowIds || {},
+      },
+    } as UseTableOptions<T> &
+      UseFiltersOptions<T> &
+      UseSortByOptions<T> &
+      UseExpandedOptions<T> &
+      UseRowSelectOptions<T> &
+      UsePaginationOptions<T>,
+    ...tableHooks
+  ) as TableInstance<T> & UseTableInstanceProps<T> & UsePaginationInstanceProps<T> & UseRowSelectInstanceProps<T>;
+
+  if (props.expandable && !props.renderExpandedComponent) {
+    throw Error('FeTable: you must provide renderExpandedComponent property if the table is expandable');
+  }
+  if (props.hasOwnProperty('sortBy') && !props.onSortChange) {
+    throw Error('FeTable: you must provide onSortChange property if sortBy is controlled');
+  }
+  if (props.hasOwnProperty('filters') && !props.onFilterChange) {
+    throw Error('FeTable: you must provide onFilterChange property if filters is controlled');
+  }
+  if (props.hasOwnProperty('pagination') && !props.pageSize) {
+    throw Error('FeTable: you must provide pageSize property if pagination enabled');
+  }
+  if (props.hasOwnProperty('onPageChange') && !props.pageCount) {
+    throw Error('FeTable: you must provide pageCount property if onPageChange is controlled');
+  }
```

Comment:
consider export it to validate function

Reply:
I will export some of the functionalities to helpers and utils